README.md

```diff
@@ -3,48 +3,124 @@ Manages the lifecycle of a NATS JetStream deployment.
 
 ## Description
 It is a standard Kubernetes operator which observes the state of NATS JetStream deployment and reconciles its state according to desired state.
-## Getting Started
-You’ll need a Kubernetes cluster to run against. You can use [k3d](https://k3d.io/) to get a local cluster for testing, or run against a remote cluster.
-**Note:** Your controller will automatically use the current context in your kubeconfig file (i.e. whatever cluster `kubectl cluster-info` shows).
 
-### Running on the cluster
-1. Install Instances of Custom Resources:
+### How it works
+
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/),
+which provide a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
```

Comment:
Maybe another sentence here, that it's based on Kubebuilder which offers commands and `make` targets which create code based on markers. And that you should refer to these when modifying stuff.